scamp/mocked_test.go

```diff
@@ -0,0 +1,38 @@
+package scamp
+
+import (
+	"context"
+	"testing"
+)
+
+func TestMockedScampRequester(t *testing.T) {
+
+	mocks := map[string]string{
+		"kids.tv.remote.fetch": `{"channel_id":123}`,
+		"godfriend.favors.ask": `{"items":[4,5,6]}`,
+		"bank.funds.withdraw":  "",
+	}
+
+	r := NewMockedScampRequester(mocks)
+
+	for act, res := range mocks {
+		mssg := NewMessage()
+		mssg.SetAction(act)
+		recv, _ := r.MakeJSONRequest(context.Background(), mssg, 10, false)
+
+		exp := NewResponseMessage()
+		exp.Write([]byte(res))
```

Comment:
do you have to go to a response message just to compare the bytes?

Reply:
mmm, you're right! I was building the response to compare them as Messages, but that did not work.
I'll simplify.